### PR TITLE
Single Laser tr3 - Tripwire

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**
positive OCB = lethal
negative OCB = not lethal (activates trigger)
color = choose in OCB menu

Laser thickness = OCB value. (For tripwire use OCB 1)

can be free rotated (needs to be activated in TE)

Slot # 426 (replaced ID_TRIPWIRE)

### CODE REVIEW

**Review by Jakub768 — APPROVED** [review]

File: TombEngine.sln [hunk 1]
```diff
@@ -15,8 +15,8 @@ Global
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
 		{15AB0220-541C-4DA1-94EB-ED3C47E4582E}.Debug|x64.ActiveCfg = Debug|x64
 		{15AB0220-541C-4DA1-94EB-ED3C47E4582E}.Debug|x64.Build.0 = Debug|x64
-		{15AB0220-541C-4DA1-94EB-ED3C47E4582E}.Debug|x86.ActiveCfg = Debug|Win32
-		{15AB0220-541C-4DA1-94EB-ED3C47E4582E}.Debug|x86.Build.0 = Debug|Win32
+		{15AB0220-541C-4DA1-94EB-ED3C47E4582E}.Debug|x86.ActiveCfg = Debug|x64
```
Comment (Lwmte) [inline TombEngine.sln:18]: Is there any reason why underlying x86 configuration was changed to x64?

File: TombEngine/Scripting/Internal/TEN/Objects/ObjectIDs.h [hunk 2]
```diff
@@ -806,6 +806,7 @@ The following constants are inside ObjID.
 	DOPPELGANGER_ORIGIN
 	CORPSE
 	WRAITH_TRAP
+	SINGLE_LASER
```
Comment (Lwmte) [inline TombEngine/Scripting/Internal/TEN/Objects/ObjectIDs.h:809]: I think this should be changed to `LASER_BEAM`

File: TombEngine/Shaders/Blending.hlsli [hunk 3]
```diff
@@ -24,182 +24,279 @@
...
+    {
+        decayFactor = (1.0f - uv.y) / 2;
+    }
+    color *= decayFactor;
+
+    color.rgb = smoothstep(ZERO, EIGHT_FIVE, color.rgb);
+    return color;
+}
+
+float4 DoLaserBeamEffect(float3 input, float4 output, float2 uv, float faceFactor, float timeUniform)
```
Comment (Lwmte) [inline TombEngine/Shaders/Blending.hlsli:205]: `Blending.hlsli` becomes a "monster file" with all new sprite render types. Can we maybe move custom drawing shaders to a new file?

File: TombEngine.sln (hunk 1 above)
Comment (Nemoel-Tomo) [reply TombEngine.sln:18]: no, this file should not be in the PR. I will remove the file from the PR again

File: TombEngine/Shaders/Blending.hlsli (hunk 3 above)
Comment (Nemoel-Tomo) [reply TombEngine/Shaders/Blending.hlsli:205]: Sorry I do not have the skills for it :(

someone else has to

**Pull request merged**